### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by sshaulnv**
mlnx_tune is only supported by machines with python2 installed on.
This commit add python3 compatibility.
The commit also change the python interpreter
that mlnx_tune use to be the first interpreter that defined
on the environment's $PATH.

Signed-off-by: Shmuel Shaul <sshaul@nvidia.com>

### CODE REVIEW

**Comment by tzafc** [comment]
Looks good. Should be applied soon, after a new product release.

File: sbin/mlnx_tune [hunk 1]
```diff
@@ -383,8 +385,18 @@ class OS:
 		if os_platform == "sunos":
 			return OS.SUNOS
 
+		using_distro = False
```
[the remark this answers is not in the document]
Comment (sshaulnv) [reply sbin/mlnx_tune]: fixed on the last commit.

**Comment by johnprakashgithub** [comment]
line number 2223and 2231, basestring is no more available in python3. use 'str' instead of 'basestring'.

File: sbin/mlnx_tune [hunk 2]
```diff
@@ -493,8 +502,8 @@ class OS:
 					return OS.XENSERVER6_5
 				if os_version == '7.1.0':
 					return OS.XENSERVER7_1
-
-		logging.warning("Unknown OS [%s,%s,%s]. Tuning might be non-optimized." % (str(platform.system()), str(platform.release()), str(platform.dist())))
+
```
Comment (tzafrir-mellanox) [inline sbin/mlnx_tune:505]: Looks good. But just please remove this extra whitespace.

**Pull request merged**